compiler/rustc_serialize/src/opaque.rs

```diff
@@ -139,26 +142,12 @@ pub struct FileEncoder {
 
 impl FileEncoder {
     pub fn new<P: AsRef<Path>>(path: P) -> io::Result<Self> {
-        const DEFAULT_BUF_SIZE: usize = 8192;
-        FileEncoder::with_capacity(path, DEFAULT_BUF_SIZE)
-    }
-
-    pub fn with_capacity<P: AsRef<Path>>(path: P, capacity: usize) -> io::Result<Self> {
-        // Require capacity at least as large as the largest LEB128 encoding
-        // here, so that we don't have to check or handle this on every write.
-        assert!(capacity >= largest_max_leb128_len());
-
-        // Require capacity small enough such that some capacity checks can be
-        // done using guaranteed non-overflowing add rather than sub, which
-        // shaves an instruction off those code paths (on x86 at least).
-        assert!(capacity <= usize::MAX - largest_max_leb128_len());
-
         // Create the file for reading and writing, because some encoders do both
         // (e.g. the metadata encoder when -Zmeta-stats is enabled)
         let file = File::options().read(true).write(true).create(true).truncate(true).open(path)?;
 
         Ok(FileEncoder {
-            buf: Box::new_uninit_slice(capacity),
+            buf: Box::new_uninit_slice(BUF_SIZE),
```

Comment:
If we are using a const-sized buffer, we could store an array instead of the slice, no need to store the length in the pointer. Something like [`.try_into().unwrap()`](https://godbolt.org/z/PTjWcaf9E) should work here.

Reply:
Not sure I follow... would this change the type of `FileEncoder::buf` to `Box<[MaybeUninit<u8>; BUF_SIZE]>`?

Reply:
Yes. (I would leave a comment on the field, but github doesn't let me :( )

Reply:
I tried this locally. It made performance marginally worse than the slice version, plus the code is a tiny bit uglier. An unnecessary word doesn't seem important, so I suggest we stick with this original version. Ok?

Reply:
It's weird that array code performs worse, I would expect it to be the same/marginally better 🤔 

But if that's the case, let's stick to the current code, sure. I'll maybe try to play with this later.